scipy/linalg/tests/test_decomp.py

```diff
@@ -1032,84 +1077,198 @@ def test_gh_5039(self):
         #
         # The matrix below is taken from the call to
         # `B = _fitpack._bsplmat(order, xk)` in interpolate._find_smoothest
-        b = np.array(
+        b = xp.asarray(
             [[0.16666667, 0.66666667, 0.16666667, 0., 0., 0.],
              [0., 0.16666667, 0.66666667, 0.16666667, 0., 0.],
              [0., 0., 0.16666667, 0.66666667, 0.16666667, 0.],
              [0., 0., 0., 0.16666667, 0.66666667, 0.16666667]])
-        svd(b, lapack_driver=self.lapack_driver)
+        svd(b, lapack_driver=lapack_driver)
 
+    @array_api_compatible
     @pytest.mark.skipif(not HAS_ILP64, reason="64-bit LAPACK required")
     @pytest.mark.slow
-    def test_large_matrix(self):
+    def test_large_matrix(self, xp):
         check_free_memory(free_mb=17000)
-        A = np.zeros([1, 2**31], dtype=np.float32)
+        A = xp.zeros([1, 2**31])
         A[0, -1] = 1
         u, s, vh = svd(A, full_matrices=False)
-        assert_allclose(s[0], 1.0)
-        assert_allclose(u[0, 0] * vh[0, -1], 1.0)
+        xp_assert_close(s[0], xp.asarray(1.0))
+        xp_assert_close(u[0, 0] * vh[0, -1], xp.asarray(1.0))
+
+
+class TestSVD_GESDD:
```

Comment:
This would be a lot nicer if we had something like the following, right?
```python
class _CheckSVD:
    self.lapack_driver = None
    test_xxx(self):
        ...

@array_api_compatible(inject_xp=True)
class TestSVD(_CheckSVD):
    # uses the default LAPACK method, `gesdd`.
    pass

class TestSVD_GESVD(_CheckSVD):
    self.lapack_driver = 'gesvd'
```

What's here now isn't too hard to understand, but it is almost 100 lines of boilerplate.